File: Frontend_EnergyApp/src/app/dtos/DeviceDto.ts
export class DeviceDto {
  id: string;
  address: string;
  description: string;
  maxHourlyEnergyConsumption: number;
  user_id: string;

  constructor(id: string, address: string, description: string, maxHourlyEnergyConsumption: number, user_id: string) {
    this.id = id;
    this.address = address;
    this.description = description;
    this.maxHourlyEnergyConsumption = maxHourlyEnergyConsumption;
    this.user_id = user_id;
  }
}

File: DeviceMicroService/src/main/java/com/example/deviceMS/dtos/deviceDTOs/DeviceDTO.java
package com.example.deviceMS.dtos.deviceDTOs;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DeviceDTO {
    private UUID id;
    private String address;
    private String description;
    private int maxHourlyEnergyConsumption;
    private UUID user_id;
}
